logic/Rule.java

```diff
@@ -241,41 +235,12 @@ public Rule rule() {
             return rule;
         }
 
-        public Conjunction conjunction() {
-            return rule().when();
-        }
-
-        public Set<Concludable> concludablesTriggeringRules(ConceptManager conceptMgr, LogicManager logicMgr) {
-            if (concludablesTriggeringRules == null) { // only acquire lock if required
-                synchronized (this) { // only compute concludables once
-                    if (concludablesTriggeringRules == null) {
-                        concludablesTriggeringRules = iterate(Concludable.create(rule.when()))
-                                .filter(c -> c.getApplicableRules(conceptMgr, logicMgr).hasNext()).toSet();
-                    }
-                }
-            }
-            return concludablesTriggeringRules;
-        }
-
-        public Set<Concludable> negatedConcludablesTriggeringRules(ConceptManager conceptMgr, LogicManager logicMgr) {
-            synchronized (this) { // can be more contentious as only used for validation
-                if (negatedConcludablesTriggeringRules == null) {
-                    negatedConcludablesTriggeringRules = concludables(rule.when().negations())
-                            .filter(c -> c.getApplicableRules(conceptMgr, logicMgr).hasNext()).toSet();
-                }
-            }
-            return negatedConcludablesTriggeringRules;
```

Comment:
Move this functionality to LogicManager & ResolvableConjunction.
ResolvableConjunction tracks the (negated) concludables, LogicManager tells us which rules they trigger.